snakeplane/plugin_factory.py

```diff
@@ -700,9 +700,12 @@ def chunk_ii_push_record(
                 ):
 
                     if current_interface.is_last_chunk == None:
-                        self._init_func(plugin)
-                        self._build_metadata(plugin)
+                        current_plugin = current_interface.parent
```

Comment:
I think you can actually just use the `plugin` object being passed to this function instead of going through the current_interface to get it.